hta/common/trace_parser.py

```diff
@@ -428,3 +436,102 @@ def parse_trace_dataframe(
             f"Parser Memory usage peak = {(peak/1024/1024):.2f} MB, current = {(current/1024/1024):.2f} MB"
         )
     return meta, df, local_symbol_table
+
+
+# --- Trace metadata reader ---
+def parse_metadata_ijson(fh) -> MetaData:
+    """
+    Parses a trace file using the `ijson` library and extracts certain high-level key-value pairs.
+    Args:
+        fh (file-like object): The trace file to parse.
+    Returns:
+        A dictionary containing the extracted metadata.
+
+    Note: We use the ijson low level API to read all dictionary
+    elements in the json upto "traceEvents", at which point we just terminate.
+    This leads to some blazingly fast metadata parsing!
```

Comment:
the use of "blazingly" here may offend some Rustaceans 🤣😉

Reply:
haha it is indeed fast :D